libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -341,6 +376,49 @@ void ControlFlowGraphBuilder::operator()(Switch const& _switch)
 {
 	yulAssert(m_currentBlock, "");
 	shared_ptr<DebugData const> preSwitchDebugData = debugDataOf(_switch);
+	auto switchBlock = m_currentBlock;
+	std::cout << "CFGB: Visiting switch expression" << std::endl;
+	StackSlot switchExpr = std::visit(*this, *_switch.expression);
```

Comment:
```suggestion
	std::cout << "CFGB: Visiting switch expression" << std::endl;
	StackSlot switchExpr = std::visit(*this, *_switch.expression);
	auto switchBlock = m_currentBlock;
```

The switch block only really starts *after* the expression is fully evaluated - like the ``if`` condition is part of the block *before* the ``if``, the switch expression is part of the block *before* the ``switch``.